### ISSUE TITLE
Call to a member function isActive() on null after update to 3.3.11 Symfony version

### ISSUE THREAD

**Issue by fjpalacios** [symfony/symfony#24934]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.3.11
| PHP version | 7.1.11
| nginx version | 1.12.2

It's the first time that I reported a possible bug to Symfony and English isn't my mother tongue, so I want to apologize if I can't explain something clearly.

After upgrading to Symfony 3.3.11 failed all the battery of tests for my project. Among other failures, the most repeated is that the status code of the client wasn't 200 but 500, I then manually checked the project and whenever there is written this code:
```php
if (!$flush) {
    $status = 'POST_EDITED_PROPERLY';            
} else {
    $status = 'POST_EDITED_ERROR';
}
$this->session->getFlashBag()->add('status', $status);
```
It's true that there is a 500 error. And the Symfony error message is: **Call to a member function isActive() on null**.

The line of the log with the critical error is:

> Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Call to a member function isActive() on null" at /Volumes/media/programación/www/Symfony/vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php line 290

The Stack Trace offered is:
> Symfony\Component\Debug\Exception\FatalThrowableError:
> Call to a member function isActive() on null
> 
>   at vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php:290
>   at Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage->getBag('flashes')
>      (vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/Session/Session.php:221)
>   at Symfony\Component\HttpFoundation\Session\Session->getBag('flashes')
>      (vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/Session/Session.php:231)
>   at Symfony\Component\HttpFoundation\Session\Session->getFlashBag()
>      (src/AppBundle/Controller/AdminController.php:201)
>   at AppBundle\Controller\AdminController->postsEditAction(object(Request), '17')
>   at call_user_func_array(array(object(AdminController), 'postsEditAction'), array(object(Request), '17'))
>      (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:153)
>   at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
>      (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68)
>   at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
>      (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:169)
>   at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
>      (web/app_dev.php:29)
>   at require('/Volumes/media/programación/www/Symfony/web/app_dev.php')
>      (vendor/symfony/symfony/src/Symfony/Bundle/WebServerBundle/Resources/router.php:42)

If I can do something more and it's within my ability just let me know.

**Comment by javiereguiluz** [comment]
@fjpalacios thanks for reporting this issue and for providing all that information!

I'd like to ask you which was the last Symfony version where your application worked. You said that the error happens after upgrading to 3.3.11. Were you using 3.3.10 before or it was an older version?

I'd also like to know if any of the external conditions changed after the upgrade (the PHP version, the php.ini config, the execution environment (docker or not), etc.) You probably didn't change anything ... but we need to be sure. Thanks!

**Comment by fjpalacios** [comment]
@javiereguiluz yes, previously this update to 3.3.11 my project was in 3.3.10. I didn't change anything at all compared to how it was before, I don't use Docker, I don't upgraded PHP or nginx, nor changed my php.ini. Just a `composer update symfony/symfony`

To ensure that the problem was the update I did `rm -rf vendor` I changed in composer.json the version of Symfony to 3.3.10 then `composer update symfony/symfony` once again and now everything worked fine.

**Comment by fjpalacios** [comment]
Perhaps not sufficiently explained what I wanted to say in the last paragraph (my English should be improved). When I returned to 3.3.10 to test everything worked fine, but if I upgrade to 3.3.11 the same 500 error is still there. Sorry if my previous message was well understood, but I preferred to be sure.

**Comment by javiereguiluz** [comment]
@fjpalacios you explained yourself perfectly: the application works with Symfony 3.3.10 but, if you upgrade to 3.3.11 and change nothing else, the application fails.

**Comment by xabbuh** [comment]
Can you share a bit more context of the code where this call is made? Where is the session attribute in your code set, for example?

**Comment by damienthiesson** [comment]
We also experienced the same issue, and had to set
`"symfony/http-foundation": "2.7.35",`
Instead of
`"symfony/http-foundation": "2.7.*"`
To move on. It is due to this commit included in the 2.7.36 release: https://github.com/symfony/http-foundation/commit/0d84932805d93dccc3a2ef1174d92548fac6c9d6

**Comment by fjpalacios** [comment]
@xabbuh my project is open source, maybe you can find useful to have all of the code. Just in case: https://github.com/sargantanacode/sargantanacode

**Comment by nicolas-grekas** [comment]
@sroze look like we messed up somewhere :) see also #24941

**Comment by javiereguiluz** [comment]
I've updated an app to 3.3.11 and I can't reproduce this or #24941. But we're not dealing with cookies in the tests of that app, which is what causes problems for @damienthiesson.

**Comment by mzampetakis** [comment]
Hello, I had a similar issue today after updating from 2.8.28 to 2.8.29. I got this error:
```
request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalErrorException: 
"Error: Call to a member function isActive() on null" at app/cache/prod/classes.php line 196 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalErrorException(code: 0): Error: Call to a member function isActive() on null at 
app/cache/prod/classes.php:196)"} []
```


This is the part of clases.php file that cause it:
`public function getBag($name)
{
if (!isset($this->bags[$name])) {
throw new \InvalidArgumentException(sprintf('The SessionBagInterface %s is not registered.', $name));
}
if ($this->saveHandler->isActive() && !$this->started) {
$this->loadSession();
} elseif (!$this->started) {
$this->start();
}
return $this->bags[$name];
}`

**Comment by cedgex** [comment]
Hello,
same problem.
updating 3.3.11 > Call to a member function isActive() on null
downgrade 3.3.10 OK
best regards

**Comment by ahinze-bln** [comment]
Hey, i use the `symfony/http-foundation` component standalone and get the same error while updating to `3.4.x-dev ce3c40e`. 

code:
```$session = new Session();
return $session;```

greetings

**Comment by albanafmeti** [comment]
I am using Laravel 5.5.
I just run `composer update` today and the same bug is happening on my side.

**Comment by sroze** [comment]
That's a good point... This will happen if your PHP session is already opened. We've decided not to call the [`setSaveHandler` method](https://github.com/symfony/symfony/blob/3.3/src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php#L384) because of [this call to `session_set_save_handler `](https://github.com/symfony/symfony/blob/3.3/src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php#L406) that isn't allowed anymore in PHP 7.2. I guess we should just stop the call to `session_set_save_handler` and not the entire `setSaveHandler` method.

**Comment by sroze** [comment]
@damienthiesson could you confirm that this issue is gone with the following PR?
https://github.com/symfony/symfony/pull/24946

If this solves it (which I guess), we can merge to the 3.x branches and this should be solved for the rest of you.

**Comment by otsoga** [comment]
I had the same problem. Downgrading back to 3.3.10 fixed it.

**Comment by sev-signal** [comment]
Hi,
I'm having the same issue using the latest version of concrete5.
I just ran a composer update during the lunch and..... "Call to a member function isActive() on null "

**Comment by fabpot** [comment]
We need people testing the fix from #24946 before merging and making new releases and confirming that this fixes this issue. Thanks for your help and sorry for the inconvenience.

**Comment by sev-signal** [comment]
This seems to work for me
